### PR TITLE
Draft: Implement data models with pydantic and template with fastapi.

### PR DESCRIPTION
**Pull request by csadorf**

### CODE REVIEW

**Comment by csadorf** [comment]
@yoavnash So, this is my first draft for this approach. I think the remaining tasks would be

- [ ] Add missing capabilities.
- [ ] Add a few tests.
- [ ] Update the getting-started app based on the template app here.

This is ready for a first rough review, I haven't tested the code yet.

I also looked into schema validation. I think we should use a [spectral ruleset](https://meta.stoplight.io/docs/spectral/ZG9jOjYx-overview) to check APIs during app registration.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                python-version: ['3.7', '3.8', '3.9', '3.10']
+                python-version: ['3.10']
```
Comment (yoavnash) [inline .github/workflows/ci.yml:30]: Won't it be somewhat problematic not supporting these older versions of Python?

Comment (csadorf) [reply .github/workflows/ci.yml:30]: Once we agree on the overall design and implementation, I am happy to bring back backwards compatibility. I think we should aim for >=3.8/ 3.9.

**Comment by pablo-de-andres** [comment]
How do we proceed with this one? Most of the functionality created is now in the standard app API repo. Should we just delete it?

**Comment by csadorf** [comment]
Superseded by #22 .

**Comment by csadorf** [comment]
Let's keep the branch as a reference.

**Pull request closed without merging**